container-engine-lib/lib/backend_impls/kubernetes/kubernetes_kurtosis_backend/shared_helpers/shared_helpers.go

```diff
@@ -682,3 +802,106 @@ func getEnclaveMatchLabels() map[string]string {
 	}
 	return matchLabels
 }
+
+func createDumpPodJob(
+	ctx context.Context,
+	kubernetesManager *kubernetes_manager.KubernetesManager,
+	namespaceName string,
+	pod apiv1.Pod,
+	enclaveOutputDirpath string,
+	resultChan chan dumpPodResult,
+) func() {
+	return func() {
+		if err := dumpPodInfo(ctx, kubernetesManager, namespaceName, pod, enclaveOutputDirpath); err != nil {
+			result := dumpPodResult{
+				podName: pod.Name,
+				err:     err,
+			}
+			resultChan <- result
+		}
+	}
+}
+
+func dumpPodInfo(
+	ctx context.Context,
+	kubernetesManager *kubernetes_manager.KubernetesManager,
+	namespaceName string,
+	pod apiv1.Pod,
+	enclaveOutputDirpath string,
+) error {
+	podName := pod.Name
+
+	// Make pod output directory
+	podOutputDirpath := path.Join(enclaveOutputDirpath, podName)
+	if err := os.Mkdir(podOutputDirpath, createdDirPerms); err != nil {
+		return stacktrace.Propagate(
+			err,
+			"An error occurred creating directory '%v' to hold the output of pod with name '%v'",
+			podOutputDirpath,
+			podName,
+		)
+	}
+
+	jsonSerializedPodSpecBytes, err := json.MarshalIndent(pod.Spec, enclaveDumpJsonSerializationPrefix, enclaveDumpJsonSerializationIndent)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred serializing the spec of pod '%v' to JSON", podName)
+	}
+	podSpecOutputFilepath := path.Join(podOutputDirpath, podSpecFilename)
+	if err := ioutil.WriteFile(podSpecOutputFilepath, jsonSerializedPodSpecBytes, createdFilePerms); err != nil {
+		return stacktrace.Propagate(
+			err,
+			"An error occurred writing the spec of pod '%v' to file '%v'",
+			podName,
+			podSpecOutputFilepath,
+		)
+	}
+
+	for _, container := range pod.Spec.Containers {
+		containerName := container.Name
+
+		// Make container output directory
+		containerLogsFilepath := path.Join(podOutputDirpath, containerName+containerLogsFilenameSuffix)
+		containerLogsOutputFp, err := os.Create(containerLogsFilepath)
+		if err != nil {
+			return stacktrace.Propagate(
+				err,
+				"An error occurred creating file '%v' to hold the logs of container with name '%v' in pod '%v'",
+				containerLogsFilepath,
+				containerName,
+				podName,
+			)
+		}
+		defer containerLogsOutputFp.Close()
+
+		containerLogReadCloser, err := kubernetesManager.GetContainerLogs(
+			ctx,
+			namespaceName,
+			podName,
+			containerName,
+			shouldFollowPodLogsWhenDumping,
+			shouldAddTimestampsWhenDumpingPodLogs,
+		)
+		if err != nil {
+			return stacktrace.Propagate(
+				err,
+				"An error occurred dumping logs of container '%v' in pod '%v' in namespace '%v'",
```

Comment:
maybe we can be more specific and say an error occurred getting logs; as this isn't dumping them just yet